tensorflow/core/platform/s3/s3_file_system_test.cc

```diff
@@ -130,6 +130,8 @@ TEST_F(S3FileSystemTest, NewReadOnlyMemoryRegionFromFile) {
 
 TEST_F(S3FileSystemTest, FileExists) {
   const string fname = TmpDir("FileExists");
+  // Ensure the file doesn't yet exist.
+  TF_ASSERT_OK(s3fs.DeleteFile(fname));
```

Comment:
Isn't DeleteFile of a file that doesn't exist supposed to return an error?

Reply:
Good question! [The API docs](https://github.com/tensorflow/tensorflow/blob/04b5c75aae4bdbdac7c713714a369f9b360daf70/tensorflow/core/platform/file_system.h#L150-L151) don't specify.

The posix filesystem looks like it will return ENOENT if the file doesn't exist, but the S3 filesystem currently treats this as success. I'd rather not change the S3 behavior, but I could future-proof this by removing the `TF_ASSERT_OK` - what do you think?

Reply:
Actually, I think we should return an error if the file doesn't exist. That's the expectation, and if we have different behavior for filesystems, then it will lead to workarounds at the call site.

Reply:
I'm worried about making a breaking change to the API here; it also worries me that it's out-of-scope for the main fix.

I can make it if you like, but it feels like it should be tracked separately.

Reply:
Also, from what I can tell, S3 doesn't communicate the delete status back from a delete. We would need to add a check for the object existing prior to deleting (fine, but not ideal).